cpp/include/cudf/strings/json.hpp

```diff
@@ -26,6 +28,76 @@ namespace strings {
  * @file
  */
 
+/**
+ * @brief Settings for `get_json_object()`.
+ */
+class get_json_object_options {
+  // allow single quotes to represent strings in JSON
+  bool allow_single_quotes = false;
+
+  // individual string values are returned with quotes stripped.
+  bool strip_quotes_from_single_strings = false;
+
+ public:
+  /**
+   * @brief Default constructor.
+   */
+  explicit get_json_object_options() = default;
+
+  /**
+   * @brief Returns true/false depending on whether single-quotes for representing strings
+   * are allowed.
+   */
+  CUDA_HOST_DEVICE_CALLABLE bool get_allow_single_quotes() const { return allow_single_quotes; }
```

Comment:
To be honest, options struct with getters/setters make no sense to me. Can we use just struct like this?
```
struct options {
  // ...
  bool allow_single_quotes = false;

  //...
  bool strip_quotes_from_single_strings = false;
};
```

Reply:
I tend to like what you're proposing, but the getter/setter model is consistent with the rest of cudf (cuio uses this a lot)